src/qt_gui/settings_dialog.h

```diff
@@ -16,7 +16,8 @@ class SettingsDialog;
 class SettingsDialog : public QDialog {
     Q_OBJECT
 public:
-    explicit SettingsDialog(QWidget* parent = nullptr);
+    explicit SettingsDialog(QWidget* parent = nullptr,
+                            std::vector<QString>* physical_devices = nullptr);
```

Comment:
```suggestion
                            std::span<const QString> physical_devices);
```

Reply:
I've tried doing that way but for some reason it doesn't compile due to an error: missing default argument on parameter 'physical_devices'
![image](https://github.com/user-attachments/assets/1bd1fd9d-f50e-4b58-9465-409e0220783b)
Not sure if this is something on my environment only

Reply:
That is because the first function argument has a default argument. Default argument are only supported at end of list. Either remove default or move std::span as first arg

Reply:
Thanks that solved the issue.